From d2a0ef6dc57133dfd4d9bb3a00bffcf901f49e49 Mon Sep 17 00:00:00 2001
From: "lauren n. liberda" <lauren@selfisekai.rocks>
Date: Thu, 27 Jun 2024 03:24:18 +0200
Subject: [PATCH] stdc++: provide constructed value to
 std::optional<T>::emplace in media_control_timeline_element.cc

workarounds https://gcc.gnu.org/bugzilla/show_bug.cgi?id=101227

Bug: 41455655
Change-Id: I9ba57b0f76d97028c93302c549fd1f12f7137824
---
 .../media_controls/elements/media_control_timeline_element.cc   | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/third_party/blink/renderer/modules/media_controls/elements/media_control_timeline_element.cc b/third_party/blink/renderer/modules/media_controls/elements/media_control_timeline_element.cc
index 1bd013a6bf62b..cef6199ea3cce 100644
--- a/third_party/blink/renderer/modules/media_controls/elements/media_control_timeline_element.cc
+++ b/third_party/blink/renderer/modules/media_controls/elements/media_control_timeline_element.cc
@@ -102,7 +102,7 @@ void MediaControlTimelineElement::UpdateAria() {
 void MediaControlTimelineElement::SetPosition(double current_time,
                                               bool suppress_aria) {
   if (is_live_ && !live_anchor_time_ && current_time != 0) {
-    live_anchor_time_.emplace();
+    live_anchor_time_.emplace(LiveAnchorTime());
     live_anchor_time_->clock_time_ = base::TimeTicks::Now();
     live_anchor_time_->media_time_ = MediaElement().currentTime();
   }
